src/v/kafka/server/handlers/fetch.cc

```diff
@@ -892,6 +1363,32 @@ kafka::fetch_plan make_simple_fetch_plan(op_context& octx) {
 }
 } // namespace testing
 
+namespace {
+ss::future<> do_fetch(op_context& octx) {
+    switch (config::shard_local_cfg().fetch_read_strategy) {
+    case model::fetch_read_strategy::polling: {
+        // first fetch, do not wait
+        co_await fetch_topic_partitions(octx).then([&octx] {
+            return ss::do_until(
+              [&octx] { return octx.should_stop_fetch(); },
+              [&octx] { return fetch_topic_partitions(octx); });
+        });
+    } break;
+    case model::fetch_read_strategy::non_polling: {
+        auto planner = make_fetch_planner<simple_fetch_planner>();
```

Comment:
nit: Wondering if this bit can be refactored so you could have a run method that just takes a T which is the plan you desire to use, instead of the different detailed logic between the switch cases above

Reply:
It's a bit difficult to do that here since the two fetch types have different logic surrounding how the fetch executor is used.